bin/server.py

```diff
@@ -64,41 +69,49 @@ async def search_opportunities(
         product_router: ProductRouter,
         search: OpportunityRequest,
         request: Request,
-    ) -> list[Opportunity]:
-        return [o.model_copy(update=search.model_dump()) for o in self._opportunities]
+    ) -> Result[list[Opportunity], Maybe[Exception]]:
+        try:
+            return Success(
+                [o.model_copy(update=search.model_dump()) for o in self._opportunities]
+            )
+        except Exception as e:
+            return Failure(e)
 
     async def create_order(
         self, product_router: ProductRouter, payload: OrderRequest, request: Request
-    ) -> Order:
+    ) -> Result[Order, Maybe[Exception]]:
```

Comment:
Same problem here:

```suggestion
    ) -> Result[Order, Exception]:
```

Reply:
So I like the pattern of `Result[Order, Maybe[Exception]]` because it allows for easy disambiguation between a successful execution (`Success`), partially defined function that's undefined for the input (`Failure(Nothing)`), and the case where there's an "real-world" error that prevented the function from being evaluated properly (`Failure(Some(Exception()))`). I think these three cases should always be allowed for, even if in the current implementation they aren't all used.